pyogrio/_io.pyx

```diff
@@ -196,27 +205,27 @@ cdef void* ogr_open(const char* path_c, int mode, char** options) except NULL:
     else:
         flags |= GDAL_OF_READONLY
 
-
     try:
         # WARNING: GDAL logs warnings about invalid open options to stderr
         # instead of raising an error
-        ogr_dataset = exc_wrap_pointer(
-            GDALOpenEx(path_c, flags, NULL, <const char *const *>options, NULL)
-        )
-
-        return ogr_dataset
+        with capture_errors() as errors:
+            ogr_dataset = GDALOpenEx(path_c, flags, NULL, <const char *const *>options, NULL)
+            return errors.check_pointer(ogr_dataset, True)
 
     except NullPointerError:
         raise DataSourceError(
-            "Failed to open dataset (mode={}): {}".format(mode, path_c.decode("utf-8"))
+            f"Failed to open dataset ({mode=}): {path_c.decode('utf-8')}"
```

Comment:
```suggestion
            f"Failed to open dataset (mode={mode}): {path_c.decode('utf-8')}"
```

Reply:
Why change this, as both are equivalent, but the existing syntax seems more concise ([ref](https://docs.python.org/3/whatsnew/3.8.html#f-strings-support-for-self-documenting-expressions-and-debugging))?

Reply:
Was not familiar with that way of using F-strings!  I thought it was a typo in converting from `.format` to F-string.  Fine to leave as is, then.